### ISSUE TITLE
Progress bar overflows because of fixed column variable

### ISSUE THREAD

**Issue by dijonkitchen** [curl/curl#3492]
### I did this
`curl --progress-bar https://www.openssl.org/source/openssl-1.0.2m.tar.gz -o tmp`
with `$COLUMNS` equal to 78
### I expected the following
Single line progress bar
### curl/libcurl version
[curl -V output]
```
curl 7.54.0 (x86_64-apple-darwin18.0) libcurl/7.54.0 LibreSSL/2.6.4 zlib/1.2.11 nghttp2/1.24.1
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz HTTP2 UnixSockets HTTPS-proxy
````

### operating system
macOS 10.14.2

### Additional information
Seems like although https://github.com/curl/curl/pull/2242#issuecomment-359168806 makes it easier to test, it doesn't work as well for users, as noted in https://github.com/Homebrew/brew/issues/5563 and https://github.com/Homebrew/brew/issues/981

A lot of users are splitting their terminal window to half the screen (around 78), so maybe a smaller default size or just always use the exact size of the terminal `$COLUMNS`?

**Comment by bagder** [comment]
You're using a curl version released *before* the overhaul we did in #2242 (which landed in 7.58.0). Since then, curl uses the environment variable first, and if that isn't set it extracts the size from the terminal using ioctl(). On mac, I believe this code runs:

https://github.com/curl/curl/blob/c497cab49b90f515dd728ce2f510d8396f91f3f6/src/tool_cb_prg.c#L221-L223

If you have a problem with a curl version after 7.58.0, then I'll be interested in learning how.

**Comment by dijonkitchen** [comment]
Yup, even with the latest brew of `curl` I get the same problem:
```
curl 7.63.0 (x86_64-apple-darwin17.7.0) libcurl/7.63.0 SecureTransport zlib/1.2.11
Release-Date: 2018-12-12
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IPv6 Largefile NTLM NTLM_WB SSL libz UnixSockets 
```

I think this is because #2242 still checks for the `$COLUMNS`, which exists as you resize the terminal. And since it's there, if it's less than 79, it'll still output multiple lines for the progress bar.

**Comment by bagder** [comment]
> still checks for the `$COLUMNS`, which exists as you resize the terminal

Yes, but it is supposed to get *updated* when you resize so it keeps having the correct number of columns visible in the terminal. When I ssh into my mac and resize the terminal, it updates `COLUMNS` to reflect the new size. I could possibly imply a bug in the terminal/shell you're using?

**Comment by bagder** [comment]
> the default 79 column width is too large for users

It's not really "the default". The variable reflects the width of your terminal! How is that wrong? Are you saying it says 79 but your terminal doesn't actually fit 79 columns?

> too large for users that are using half their screen for their terminal

This sounds like a different problem now. You mean you rather have it *shorter* than COLUMNS?

> If it's set lower, I think less people would see this problem.

I don't understand why. The progress bar should not go beyond a single line if COLUMNS is just set correctly to the width of the terminal (or shorter).

Can you please help us understand exactly when and how this causes a problem?

**Comment by dijonkitchen** [comment]
My apologies, this does work for the latest `curl`, but not `brew` for some reason even though I thought homebrew would use the same `curl` for this.
